web_page_archive.install

```diff
@@ -225,3 +320,115 @@ function web_page_archive_update_8008() {
   $config->set('system.npm_path', '');
   $config->save();
 }
+
+/**
+ * Adds wpa_run_comparison and wpa_run_comparison_revision tables.
+ */
+function web_page_archive_update_8009() {
+  $entity_type_manager = \Drupal::entityTypeManager();
+  $update_manager = \Drupal::entityDefinitionUpdateManager();
+  $definition = $entity_type_manager->getDefinition('wpa_run_comparison');
```

Comment:
This is problematic similar to this: https://www.drupal.org/project/drupal/issues/2803179#comment-11644435

Basically we can't call getDefinition() on the entity type manager here as the table doesn't exist yet.   I will evaluate the best solution for this and proceed from there.

Reply:
Okay I've resolved this in the most recent commit. 

Based on feedback from Wim Leers in that referenced issue above, instead of using entity type manager to retrieve the definition in the update hook, you can just instantiate the entity type directly by copying the entity's annotation definition into the constructor. 

So in this case:

```php
/**
 * Defines the run comparison entity.
 *
 * @ingroup web_page_archive
 *
 * @ContentEntityType(
 *   id = "wpa_run_comparison",
 *   label = @Translation("Web Page Archive Run Comparison"),
 *   handlers = {
 *     "storage" = "Drupal\web_page_archive\Entity\Sql\RunComparisonStorage",
 *     "list_builder" = "Drupal\Core\Entity\EntityListBuilder",
 *     "view_builder" = "Drupal\Core\Entity\EntityViewBuilder",
 *     "form" = {
 *       "delete" = "Drupal\web_page_archive\Form\RunComparisonDeleteForm",
 *     },
 *   },
 *   base_table = "wpa_run_comparison",
 *   revision_table = "wpa_run_comparison_revision",
 *   revision_data_table = "wpa_run_comparison_field_revision",
 *   admin_permission = "administer web page archive",
 *   fieldable = TRUE,
 *   entity_keys = {
 *     "id" = "id",
 *     "revision" = "vid",
 *     "label" = "name",
 *     "uuid" = "uuid",
 *     "uid" = "user_id",
 *     "langcode" = "langcode",
 *     "status" = "status",
 *   },
 *   field_ui_base_route = "wpa_run_comparison.settings"
 * )
 */
```

The annotation gets translated into the following PHP:

```php
  $update_manager = \Drupal::entityDefinitionUpdateManager();
  $definition = new ContentEntityType([
    'id' => 'wpa_run_comparison',
    'label' => 'Web Page Archive Run Comparison',
    'handlers' => [
      'storage' => 'Drupal\web_page_archive\Entity\Sql\RunComparisonStorage',
      'list_builder' => 'Drupal\Core\Entity\EntityListBuilder',
      'view_builder' => 'Drupal\Core\Entity\EntityViewBuilder',
      'form' => [
        'delete' => 'Drupal\web_page_archive\Form\RunComparisonDeleteForm',
      ],
    ],
    'base_table' => 'wpa_run_comparison',
    'revision_table' => 'wpa_run_comparison_revision',
    'revision_data_table' => 'wpa_run_comparison_field_revision',
    'admin_permission' => 'administer web page archive',
    'fieldable' => TRUE,
    'entity_keys' => [
      'id' => 'id',
      'revision' => 'vid',
      'label' => 'name',
      'uuid' => 'uuid',
      'uid' => 'user_id',
      'langcode' => 'langcode',
      'status' => 'status',
    ],
    'field_ui_base_route' => 'wpa_run_comparison.settings',
  ]);
  $update_manager->installEntityType($definition);
```